usr/share/device-quirks/scripts/override_dsdt

```diff
@@ -17,9 +17,48 @@ ACPI_BUILD_DIR="/tmp/kernel/firmware/acpi"
 DEPLOYMENT_DSDT_PATH="${SUBVOL}/usr/lib/firmware/dsdt/${1}"
 BOOTLOADER_CONFIG="${MOUNT_PATH}/boot/loader/entries/frzr.conf"
 ACPI_OVERRIDE_DEVICE="${split_arr[0]}_acpi_override"
+OVERRIDE_LOG="/etc/device-quirks/dsdt_override.log"
+BIOS_DATE="$(cat "/sys/class/dmi/id/bios_date")"
+BIOS_RELEASE="$(cat "/sys/class/dmi/id/bios_release")"
+BIOS_VENDOR="$(cat "/sys/class/dmi/id/bios_vendor")"
+BIOS_VERSION="$(cat "/sys/class/dmi/id/bios_version")"
 
-if ! grep -q "${ACPI_OVERRIDE_DEVICE}" ${BOOTLOADER_CONFIG}; then
+# Grab previous stored values.
+source $OVERRIDE_LOG
+
+# Before doing anything we must check the checksum of the device in use against the known pool of patchable firmwares
+cat /sys/firmware/acpi/tables/DSDT > system-dumped-dsdt.dat
+# Dump DSDT from the system
+iasl -d system-dumped-dsdt.dat
+iasl -tc system-dumped-dsdt.dat
+
+dsl_file="system-dumped-dsdt.dsl"
+
+# Use sed to extract the checksum value from the DSL file
+checksum=$(sed -n 's/.*Checksum[[:space:]]\+\(0x[[:xdigit:]]\+\).*/\1/p' "$dsl_file")
+checksum2=$(sed -n 's/.*Checksum[[:space:]]\+\(0x[[:xdigit:]]\+\).*/\1/p' "${SUBVOL}/usr/lib/firmware/dsdt/${1}")
+apply_patch=0
+
+echo "Checksum value from system dump: $checksum"
+echo "Checksum of shipped DSDT: $checksum2"
+
+# Compare the checksums
+if [ "$checksum" = "$checksum2" ]; then
+  echo "Checksums match! We are able to patch the firmware"
 
+if [ "$BIOS_DATE" == "$LAST_BIOS_DATE" ] && \
```

Comment:
should this be `elif`

these if statements don't seem to have the correct syntax